index.js

```diff
@@ -23,71 +29,139 @@ async function main() {
-		console.log(`✔ Unzip Node.js`.green);
-		console.log("\nStep 8: Deleting Unziped Node.js File");
-		fs.unlinkSync(path.join(`node-v${version}-linux-x64.tar.xz`));
-		console.log(`✔ Deleted Unziped Node.js File`.green);
-	} catch (e) {
-		console.error(`✖ Error`.red);
-		throw e;
-	}
-}
-main();
+	copyTemplates([nodeRunnerFile, 'bootstrap'])
+	.then(() => replaceBootstrapStrings(version, nodeRunnerFile))
+	.then(bootstrap => writeBootstrapToDisk(bootstrap))
+	.then(() => makeExecutable(['bootstrap', nodeRunnerFile]))
+	.then(() => downloadFile(downloadURL, downloadName))
+	.then((fileName) => unzip(fileName))
+	.then(() => createRuntimeZip(nodeFolder, runtimeName))
+	.then(() => houseKeeping())
+	.then(() => {
+		return console.log(`\n\nYour node runtime is in node-v${version}-linux-x64-lambda-runtime.zip\n`.yellow);
+	})
+	.catch(err => {
+		console.log('Oh snap! Got an error!'.red);
+		throw err;
+	})
+};
 
-
-async function downloadFile(url, path) {
-	const response = await axios({
-		method: 'GET',
-		url,
-		responseType: 'stream'
+const houseKeeping = () => {
+	return new Promise((resolve, reject) => {
+		rimraf.sync(nodeFolder, { glob: false });
+		fs.unlinkSync('bootstrap');
+		fs.unlinkSync('node_runtime.js');
+		console.log('✔ Cleaned up artifacts.'.green);
+		return resolve();
 	});
+};
 
-	response.data.pipe(fs.createWriteStream(path));
+const createRuntimeZip = (nodeDir, runtimePath) => {
+	return new Promise((resolve, reject) => {
+		const archive = archiver('zip', { zlip: { level: 9 } });
+		const output = fs.createWriteStream(`${runtimePath}.zip`);
+
+		archive.pipe(output);
+		archive.file('bootstrap', { name: 'bootstrap' });
+		archive.file('node_runtime.js', { name: 'node_runtime.js' });
+		archive.file(`${nodeDir}/bin/node`, { name: 'node' });
+		archive.finalize();
+	
+		output.on('close', () => {
+			return resolve(archive)
+		});
+		archive.on('error', err => reject(err));
+	});
+};
 
+const copyTemplates = files => {
 	return new Promise((resolve, reject) => {
-		response.data.on('end', () => {
-			resolve();
+		files.map(file => {
+			fs.copyFile(path.join(__dirname, 'templates', file), file, () => {
+				console.log(`✔ Copied ${file} template file to project directory`.green);
+			}, err => {
+				reject(err);
+			});
 		});
+		
+		resolve();
+	});
+};
+
+function replaceBootstrapStrings(version, nodeRunnerFile) {
+	return new Promise((resolve, reject) => {
+		fs.readFile(`bootstrap`, 'utf8', (err, contents) => {
+			if (err) reject(err);
+			contents = contents
+				.replace(/{{NODE_VERSION}}/g, version)
+				.replace(/{{NODE_RUNNER_FILE}}/g, nodeRunnerFile);
+			console.log(`✔ Replaced template strings for bootstrap file with version ${version} and runner ${nodeRunnerFile}`.green);
 
-		response.data.on('error', () => {
-			reject();
+			resolve(contents);
 		});
+	})
+};
+
+const writeBootstrapToDisk = contents => {
+	fs.writeFileSync('bootstrap', contents, () => {
```

Comment:
Is this really supposed to be sync when we pass in a callback?

Reply:
Should not be sync.